docs/dev_guide/diagnostic_ref/dpct1039.rst

```diff
@@ -31,4 +31,52 @@ Suggestions to Fix
 
 If the first parameter of an atomic function points to a local memory address
 space, replace the atomic function name with an atomic function name that
-includes the template parameters.
\ No newline at end of file
+includes the template parameters.
+
+For example, this original CUDA\* code:
+
+.. code-block:: cpp
+   :linenos:
+
+    __device__ void d(int* s) {
+      atomicAdd(s, 1);
+    }
+    
+    __global__ void k() {
+      __shared__ int shared;
+      d(&shared);
+    }
+
+results in the following migrated SYCL code:
+
+.. code-block:: cpp
+   :linenos:
+
+    void d(int* s) {
+      /*
+      DPCT1039:0: The generated code assumes that "s" points to the global memory
+      address space. If it points to a local memory address space, replace
+      "atomic_fetch_add" with
+      "atomic_fetch_add<sycl::access::address_space::local_space>".
+      */
+      dpct::atomic_fetch_add(s, 1);
+    }
+    
+    void k(int &shared) {
+    
+      d(&shared);
+    }
+
+which is rewritten to:
+
+.. code-block:: cpp
+   :linenos:
+
+    void d(int* s) {

```

Comment:
consider to make d() as a template function,  make it can run well for both global_space and local_space call.

Reply:
done